src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/utils/HttpHelper.java

```diff
@@ -277,24 +276,16 @@ private static void addCrumbToConnection(HttpURLConnection connection, BuildCont
 	 * ATTENTION: TRUSTING ALL CERTIFICATES IS VERY DANGEROUS AND SHOULD ONLY BE USED IF YOU KNOW WHAT YOU DO!
 	 * @param context The build context
 	 * @param url The url to the remote build
-	 * @param overrideAuth
+	 * @param auth
 	 * @return An authorized connection with or without a NaiveTrustManager
 	 * @throws IOException
 	 */
-	private static URLConnection getAuthorizedConnection(BuildContext context, URL url, Auth2 overrideAuth)
+	private static URLConnection getAuthorizedConnection(BuildContext context, URL url, Auth2 auth)
 			throws IOException {
 		URLConnection connection = context.effectiveRemoteServer.isUseProxy() ? ProxyConfiguration.open(url)
 				: url.openConnection();
 
-		Auth2 serverAuth = context.effectiveRemoteServer.getAuth2();
-
-		if (overrideAuth != null && !(overrideAuth instanceof NullAuth)) {
-			// Override Authorization Header if configured locally
-			overrideAuth.setAuthorizationHeader(connection, context);
-		} else if (serverAuth != null) {
-			// Set Authorization Header configured globally for remoteServer
-			serverAuth.setAuthorizationHeader(connection, context);
-		}
+		auth.setAuthorizationHeader(connection, context);
```

Comment:
Simpler to calculate the effective authn once, at the entry point to this class.